### PR TITLE
Add middleware to check for published homepage or 404

### LINKED ISSUE #253: Language redirection active even though language not translated
Currently when visiting https://guide.wagtail.org/ it redirects to the browser preferred language and shows that language in the dropdown even though that language is not available yet. Which is a bit confusing.
E.g. preferred browser language German redirects to https://guide.wagtail.org/de-latest/ even though the content is not ready yet and displays "Deutsch (latest)" in the language drop down. See screenshot below.

Also, English is not available in the language selection list.

It would be good to disable languages that are not yet translated to avoid confusion I think?

image

### PR DESCRIPTION
**Pull request by allcaps**
Fixes #253

### CODE REVIEW

File: apps/core/middleware.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+from apps.core.models import HomePage
+
+
+class ValidateLocaleMiddleware:
+    def __init__(self, get_response):
+        self.get_response = get_response
+
+    def __call__(self, request):
+        if lang_code := get_language_from_path(request.path_info):
+            print(lang_code)
```
Comment (allcaps) [inline apps/core/middleware.py:13]: Err, print

File: apps/core/middleware.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+from apps.core.models import HomePage
+
+
+class ValidateLocaleMiddleware:
+    def __init__(self, get_response):
+        self.get_response = get_response
+
+    def __call__(self, request):
+        if lang_code := get_language_from_path(request.path_info):
```
Comment (allcaps) [inline apps/core/middleware.py:12]: Since we are after `LocalMiddleware, we can use `django.utils.translation.get_language()` to get the current language.

**Comment by allcaps** [comment]
@saevarom Thank you for the review! I've updated this PR and switched to the `get_language()` method.

**Review by saevarom — APPROVED** [review]
Tested again, looks good

**Pull request merged**